src/lib/y2network/interface_config_builder.rb

```diff
@@ -53,10 +55,20 @@ def [](key)
     end
 
     def save
-      return if driver.empty?
+      if !driver.empty?
+        Yast::LanItems.setDriver(driver)
+        Yast::LanItems.driver_options[driver] = driver_options
+      end
+
+      # create new instance as name can change
+      firewall_interface = Y2Firewall::Firewalld::Interface.new(name)
+      if Y2Firewall::Firewalld.instance.installed?
+        Yast::LanItems.firewall_zone = firewall_zone
+        # TODO: should change only if different, but maybe firewall_interface responsibility?
```

Comment:
In fact it already takes that responsibility.

https://github.com/yast/yast-yast2/blob/master/library/network/src/lib/y2firewall/firewalld/relations.rb#L194
https://github.com/yast/yast-yast2/blob/master/library/network/src/lib/y2firewall/firewalld/interface.rb#L105

So, maybe not needed at all.

Reply:
not sure, as that zone=  at first delete it and then add it again. Not sure if it is problem or not.